### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by oyelakinG9**
# BOOKSTORE:  Connect to API 🎉🎉

Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
Refactor  add book and remove book features to persist changes in the server.
 - Refactor the redux code of your application using middleware to make async requests to the API.
 - Don't do the changes using only React state.
Add book and remove book should work in the same way after the refactor.
No styling is required.

### CODE REVIEW

File: src/Components/BookList.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchBooks } from '../redux/Books/Books';
```
Comment (omar-labana) [inline src/Components/BookList.js:3]: - [ ] There is an error about `keys` when looping/mapping over an array(Booklist) in the console, kindly fix them.

Comment (omar-labana) [inline src/Components/BookList.js:3]: - [ ] There is an error about `keys` when looping/mapping over an array in the console, kindly fix them.

Comment (omar-labana) [inline src/Components/BookList.js:3]: - [ ] There is an error about `keys` when looping/mapping over an array in the console, kindly fix them.

File: src/redux/Books/Books.js [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
   books: [],
 };
+
+// FUNCTIONS
+const postBook = (book) => {
+  axios.post(`${API_URL}`, book);
+};
+const deleteBook = (id) => {
+  axios.delete(`${API_URL}${id}`);
+};
```
Comment (omar-labana) [inline src/redux/Books/Books.js]: - [ ] The delete request is failing with a 404 error, please investigate the issue.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @oyelakinG9, 

Excellent work so far connecting your app to the API! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/Books/Books.js (hunk 2 above)
Comment (oyelakinG9) [reply src/redux/Books/Books.js]: Thanks @omar-labana

**Review by AshakaE — APPROVED** [review]
### Hi @oyelakinG9   🙋‍♀️, 

# STATUS APPROVED ✔

   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**